src/common/ceph_context.cc

```diff
@@ -285,11 +294,11 @@ class CephContextObs : public md_config_obs_t {
                           const std::set <std::string> &changed) override {
     if (changed.count(
 	  "enable_experimental_unrecoverable_data_corrupting_features")) {
-      ceph_spin_lock(&cct->_feature_lock);
+      ceph::spin_lock(&cct->_feature_lock);
       get_str_set(
```

Comment:
I'd be tempted to replace all of these calls to direct lock/unlock with scoped lockers.